contracts/contracts/strategies/ThreePoolStrategy.sol

```diff
@@ -200,12 +200,15 @@ contract ThreePoolStrategy is InitializableAbstractStrategy {
         // should always stake the full balance in the Gauge, but include for
         // safety
         (, , uint256 totalPTokens) = _getTotalPTokens();
-        balance = 0;
-        if (totalPTokens > 0) {
-            balance += ICurvePool(platformAddress).calc_withdraw_one_coin(
-                totalPTokens,
-                poolCoinIndex
-            );
+        ICurvePool curvePool = ICurvePool(platformAddress);
+
+        uint256 pTokenTotalSupply = IERC20(assetToPToken[assetsMapped[0]])
```

Comment:
Since we are using assetsMapped[0] rather than _asset, do we care that checkBalance would return an invalid result if it would be called with the address of an asset that this strategy does not support (but that Curve supports so that line 208 below does return a non zero value).

Reply:
The Vault requires the asset to be supported before it calls checkBalance, but you could cause a revert by calling this method directly on the strategy. I cleaned it up a bit with a nicer error message, please take a look :)